File: iAsync.ObjC/lib/JFFUI/UITableView/UITableView+CellsSelections.m
#import "UITableView+CellsSelections.h"

#import "UITableView+BlocksAdditions.h"

@implementation UITableView (CellsSelections)

- (void)selectAllRowsAnimated:(BOOL)animated
{
    [self enumerateAllIndexPaths:^(NSIndexPath *indexPath) {
        
        [self selectRowAtIndexPath:indexPath animated:animated scrollPosition:(UITableViewScrollPositionNone)];
    }];
}

- (void)deselectAllRowsAnimated:(BOOL)animated
{
    [self enumerateAllIndexPaths:^(NSIndexPath *indexPath) {
        
        [self deselectRowAtIndexPath:indexPath animated:animated];
    }];
}

@end

File: iAsync.Swift/lib/JUI/JUI/UITableView/UITableView+CellsSelections.swift
import Foundation

public extension UITableView {
    
    func selectAllRowsAnimated(animated: Bool) {

        enumerateAllIndexPaths( { (indexPath: NSIndexPath) -> () in
            
            self.selectRowAtIndexPath(indexPath, animated:animated, scrollPosition:.None)
        })
    }
    
    func deselectAllRowsAnimated(animated: Bool) {
    
        enumerateAllIndexPaths( { (indexPath: NSIndexPath) -> () in
            
            self.deselectRowAtIndexPath(indexPath, animated:animated)
        })
    }
}
